### ISSUE TITLE
about submission

### ISSUE THREAD

**Issue by wolfworld6** [happyharrycn/actionformer_release#29]
thx for your work，could you tell me nore details about “score” in submission format？

**Comment by happyharrycn** [comment]
I am not sure I understand you question here. If you are talking about the external scores, they are from an action recognition model and not necessary for most of the datasets.

**Comment by wolfworld6** [comment]
> I am not sure I understand you question here. If you are talking about the external scores, they are from an action recognition model and not necessary for most of the datasets.

I mean when you run on the test dataset？

**Comment by wolfworld6** [comment]
> I am not sure I understand you question here. If you are talking about the external scores, they are from an action recognition model and not necessary for most of the datasets.

could you share me the test dataset json file， just like the epic_kitchens_100_noun.json？

**Comment by happyharrycn** [comment]
To run on your own test set with unlabeled videos, you will need to 
- extract video features the same way as the training set
- create a json file following the AcitivtyNet format with each annotation as an empty list
- copy the configfile  used for training, and modify the fields "json_file" / " feat_folder" to point to the new json file and video features
- if "external_score_file" was specified in the original config, remove the filed and set "multiclass_nms" to True (disable the use of external classification scores)
- modify the eval.py to save the results into the you desired format.

**Comment by wolfworld6** [comment]
> * multiclass_nms
thx very much。
sorry to bother you, how do i get the segment，Because both nouns and verbs have this parameter predicted，thx
![image](https://user-images.githubusercontent.com/18051979/169495965-df375b07-e4e7-4a06-8fb8-e6c821795166.png)

**Comment by tzzcl** [comment]
The segment means the real action starting and ending time (in seconds).

You may want to refer to the https://github.com/happyharrycn/actionformer_release/blob/main/libs/modeling/meta_archs.py#L643 for more details about how we convert the raw model outputs into final predictions.

**Comment by wolfworld6** [comment]
> The segment means the real action starting and ending time (in seconds).
> 
> You may want to refer to the https://github.com/happyharrycn/actionformer_release/blob/main/libs/modeling/meta_archs.py#L643 for more details about how we convert the raw model outputs into final predictions.

I mean to submit the results in this web，https://codalab.lisn.upsaclay.fr/，Nouns and verbs submitted separately？

**Comment by tzzcl** [comment]
You may need to refer to the specfic submission format for details. I think they should be submitted separately.

**Comment by wolfworld6** [comment]
> You may need to refer to the specfic submission format for details. I think they should be submitted separately.

 I can't open this URL about the specfic submission formats，could you？

**Comment by tzzcl** [comment]
I think you may need to email the competition organizer to report this issue.

**Comment by happyharrycn** [comment]
If you are looking for combining noun and verb predictions, you might want check the discussion in issue #7. If you have trouble for EPIC-Kitchens data format, please contact the competition organizer.

**Comment by wolfworld6** [comment]
> If you are looking for combining noun and verb predictions, you might want to check the discussion in issue #7. If you have questions for EPIC-Kitchens data format, please contact the competition organizer.

I am so confused ，both nouns and verbs predict segment time, how do I combine them to generate actions？

**Comment by tzzcl** [comment]
For this part, you can have various choices to fuse these results.

For example, you can get the noun/verb predictions for the same point, then take the noun or verb segment predictions as the final segment for this point, or you can simply take the average of these two segments. You may need to modify the code a little bit.

We may update this part shortly.

**Comment by wolfworld6** [comment]
> 

thxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx ! amazing！

**Comment by happyharrycn** [comment]
Mark as resolved.
